services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/PushMonitorUtils.java

```diff
@@ -23,11 +23,159 @@ public class PushMonitorUtils {
   private static final Logger LOGGER = LogManager.getLogger(PushMonitorUtils.class);
 
   /**
+   * This method checks Da Vinci client push status for the target version from push status store and compute a final
+   * status. It will try to get an aggregated status from version level status key first; if not found, it will fall
```

Comment:
Can we update the javadoc to make it more clear as it is not a straight fallback, but more like a merge all the time until we fully deprecate the partition-level status report, right?

Reply:
Right, I missed that. Updated in the latest diff.